assets/js/src/welcome_wizard/steps_controller.jsx

```diff
@@ -8,57 +8,42 @@ import WelcomeWizardEmailCourseStep from './steps/email_course_step.jsx';
 import WelcomeWizardUsageTrackingStep from './steps/usage_tracking_step.jsx';
 import WelcomeWizardWooCommerceStep from './steps/woo_commerce_step.jsx';
 
-class WelcomeWizardStepsController extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      stepsCount: window.is_woocommerce_active ? 4 : 3,
-      shouldSetSender: !window.is_mp2_migration_complete,
-      loading: false,
-      sender: window.sender_data,
-      replyTo: window.reply_to_data,
-    };
-
-    this.finishWizard = this.finishWizard.bind(this);
-    this.updateSettings = this.updateSettings.bind(this);
-    this.activateTracking = this.activateTracking.bind(this);
-    this.updateSender = this.updateSender.bind(this);
-    this.updateReplyTo = this.updateReplyTo.bind(this);
-    this.submitSender = this.submitSender.bind(this);
-    this.showWooCommerceStepOrFinish = this.showWooCommerceStepOrFinish.bind(this);
-    this.componentDidUpdate();
-  }
-
-  componentDidUpdate() {
-    const step = parseInt(this.props.match.params.step, 10);
-    if (step > this.state.stepsCount || step < 1) {
-      this.props.history.push('/steps/1');
+const WelcomeWizardStepsController = (props) => {
+  const [stepsCount] = useState(window.is_woocommerce_active ? 4 : 3);
```

Comment:
We don't need to define a state for constants. I would define a normal variable instead:
```js
const stepsCount = window.is_woocommerce_active ? 4 : 3;
```

Reply:
Good find. It was originally part of a state so I just moved them but they don't have to be in state at all.